cpp/log/test.cpp

```diff
@@ -61,4 +68,76 @@ void test() {
 	t1.join();
 }
 
+/**
+ * Creates a FileSink to also write log Messages
+ * into /tmp/openage-log-test.
+ * Only Messages with a level equals or higher
+ * than level::err will be printed in that file.
+ */
+void file_log_sink () {
+	const char *filelog_name = "/tmp/openage-log-test";
+	TestLogSource logger;
+	FileSink filelog(filelog_name, true);
+
+	filelog.loglevel = log::level::err;
+
+	// filelog should ignore these Messages
+	logger.log(MSG(dbg) << "This message should not appear in openage-log-test");
+	logger.log(MSG(info) << "This message should not appear in openage-log-test");
+	logger.log(MSG(warn) << "This message should not appear in openage-log-test");
+
+	// filelog should write these Messages
+	logger.log(MSG(err) << "This message should appear in openage-log-test");
+	logger.log(MSG(crit) << "This message should appear in openage-log-test");
+
+	std::string line;
+	std::ifstream infile(filelog_name);
```

Comment:
@mic-e it does allow opening the file but it needs to have the "share" access. Depending on how the file was opened for writing (with or without access sharing for reading or writing), the file may or may not be accessible for reading.